crates/xflags-macros/src/emit.rs

```diff
@@ -380,101 +400,63 @@ fn emit_default_transitions(buf: &mut String, cmd: &ast::Cmd) {
-        empty_help = false;
-        let idens = cmd.all_identifiers().cloned().collect::<Vec<_>>().join(" | ");
-        w!(buf, "{}{}\n", prefix, idens);
+    if cmd.has_subcommands() {
+        w!(help_buf, " <COMMAND>")
     }
     if let Some(doc) = &cmd.doc {
-        empty_help = false;
-        write_lines_indented(buf, doc, 2);
-    }
-    let indent = if prefix.is_empty() { "" } else { "  " };
-
-    let args = cmd.args_with_default();
-    if !args.is_empty() {
-        if !empty_help {
-            blank_line(buf);
-        }
-        empty_help = false;
-        w!(buf, "{}ARGS:\n", indent);
-
-        let mut blank = "";
-        for arg in &args {
-            w!(buf, "{blank}");
-            blank = "\n";
-
-            let (l, r) = match arg.arity {
-                ast::Arity::Optional => ("[", "]"),
-                ast::Arity::Required => ("<", ">"),
-                ast::Arity::Repeated => ("<", ">..."),
-            };
-            w!(buf, "    {l}{}{r}\n", arg.val.name);
-            if let Some(doc) = &arg.doc {
-                write_lines_indented(buf, doc, 6)
-            }
+        w!(help_buf, "\n\n{}\n", doc);
+    }
+    if !cmd.args_with_default().is_empty() {
+        w!(help_buf, "\nArguments:\n");
+        for arg in cmd.args_with_default() {
+            let (l, r) = arg.arity.brackets();
+            let pre_doc = format!("{l}{}{r}", arg.val.name);
+            w!(help_buf, "  {:<20} {}\n", pre_doc, arg.doc.as_deref().unwrap_or(""));
         }
     }
-
-    let flags = cmd.flags_with_default();
-    if !flags.is_empty() {
-        if !empty_help {
-            blank_line(buf);
-        }
-        w!(buf, "{indent}OPTIONS:\n");
-
-        let mut blank = "";
-        for flag in &flags {
-            w!(buf, "{blank}",);
-            blank = "\n";
-
+    if !cmd.flags_with_default().is_empty() {
+        w!(help_buf, "\nOptions:\n");
+        for flag in cmd.flags_with_default() {
             let short = flag.short.as_ref().map(|it| format!("-{it}, ")).unwrap_or_default();
             let value = flag.val.as_ref().map(|it| format!(" <{}>", it.name)).unwrap_or_default();
-            w!(buf, "    {short}--{}{value}\n", flag.name);
-            if let Some(doc) = &flag.doc {
-                write_lines_indented(buf, doc, 6);
-            }
+            let pre_doc = format!("{short}--{}{value}", flag.name);
+            w!(help_buf, "  {:<20} {}\n", pre_doc, flag.doc.as_deref().unwrap_or(""));
         }
     }
-
-    let subcommands = cmd.named_subcommands();
-    if !subcommands.is_empty() {
-        if prefix.is_empty() {
-            blank_line(buf);
-            w!(buf, "SUBCOMMANDS:");
-        }
-
-        let prefix = format!("{}{} ", prefix, cmd.name);
-        for sub in subcommands {
-            blank_line(buf);
-            blank_line(buf);
-            help_rec(buf, &prefix, sub);
-        }
+    w!(help_buf, "\nCommands:");
+    for subcommand in &cmd.subcommands {
```

Comment:
I think here we should continue to use `cmd.named_subcommands()` so the `default cmd` is not listed.

From doc:
> To make subcommand name optional use the **default** keyword to mark a subcommand to select if no subcommand name is passed. The name of the default subcommand affects only the name of the generated Rust struct, it can’t be specified explicitly on the command line.